cmdstanpy/cmdstan_args.py

```diff
@@ -381,9 +381,12 @@ def validate(self) -> None:
                             ' found {}'.format(self.seed[i])
                         )
 
-        if self.data is not None:
+        if isinstance(self.data, str):
             if not os.path.exists(self.data):
                 raise ValueError('no such file {}'.format(self.data))
+        elif not isinstance(self.data, dict):
```

Comment:
don't you need to update line 293  - type hint on data - else how will self.data be a dict?